### ISSUE TITLE
Session issue

### ISSUE THREAD

**Issue by afrasiyabhaider** [realrashid/sweet-alert#81]
- Package Version: 3.1.4
- Laravel Version: 7.0

when I goes back from edit page without saving data it shows the message of last action which we have performed and displayed in toast.

It is not only on edit page it is on every page

**Comment by afrasiyabhaider** [comment]
@realrashid

**Comment by afrasiyabhaider** [comment]
I've tried `Session::remove("alert");` but it doesn't work when clicked on back button of browser

**Comment by realrashid** [comment]
Hey @afrasiyabhaider ,

Please share complete step with code snippets 😊

That will help me to understand and debug what's wrong!

**Comment by realrashid** [comment]
Duplicate of #79

**Comment by realrashid** [comment]
Hey Dear,

I have deep investigated on the issue i come up with statement.

The session will exit until the next page reload.

**Comment by rafaellannes** [comment]
Hello
The same issue here... any way to resolve?

Thanks

**Comment by afrasiyabhaider** [comment]
@realrashid is there any way to resolve this issue....??

When I update a record it works fine and displays a message which is good... but when I click on the edit button for next time and press the back button without updating a record.. It again shows me a message which was previously displayed..... This shouldn't happen... Can you suggest me a solution?

**Comment by realrashid** [comment]
Right now i haven't come up with a solution.

But i am no it.

**Comment by afrasiyabhaider** [comment]
I think you should use session()->forget(key_here); .This shouldwork please give it a try @realrashid

**Comment by trollfalgar** [comment]
I have the same problem. Start recently.

**Comment by cyrilxx** [comment]
any update on this ?

**Comment by iamarbabhussain** [comment]
Any update to this?

**Comment by iamarbabhussain** [comment]
@afrasiyabhaider @realrashid

**Comment by realrashid** [comment]
Not Yet... 😭

**Comment by afrasiyabhaider** [comment]
How I can help with this? @realrashid

**Comment by abdelrhmanMohmaed** [comment]
please how can fix this issue #79 .

**Comment by iamarbabhussain** [comment]
AFAIK, its because of caching - we can disable it so everytime we get brand new page from server, no?
This way recent toasts would also go.

**Comment by anggayudap** [comment]
any update on this ?

**Comment by anggayudap** [comment]
> this solution was working with me at "uxweb/ sweet-alert" i have tried to makes it fit with this package but i can not so i post it here maybe any one else can make it fit @realrashid
> 
> <script> if (window.performance) { var navEntries = window.performance.getEntriesByType('navigation'); if (navEntries.length > 0 && navEntries[0].type === 'back_forward') { console.log('As per API lv2, this page is load from back/forward'); } else if (window.performance.navigation && window.performance.navigation.type == window.performance.navigation.TYPE_BACK_FORWARD) { console.log('As per API lv1, this page is load from back/forward'); } else { console.log('This is normal page load'); @if (Session::has('sweet_alert.alert')) swal( {!! Session::get('sweet_alert.alert') !!} ); {{ Session::forget('sweet_alert.alert') }} // This will forget the alert data after displaying it :) @endif } } else { console.log("Unfortunately, your browser doesn't support this API"); } </script>

thank you man, your code works on me

**Comment by karim20160300** [comment]
@anggayudap works for "uxweb/ sweet-alert" or this package ?

**Comment by karim20160300** [comment]
> > this solution was working with me at "uxweb/ sweet-alert" i have tried to makes it fit with this package but i can not so i post it here maybe any one else can make it fit @realrashid
> > <script> if (window.performance) { var navEntries = window.performance.getEntriesByType('navigation'); if (navEntries.length > 0 && navEntries[0].type === 'back_forward') { console.log('As per API lv2, this page is load from back/forward'); } else if (window.performance.navigation && window.performance.navigation.type == window.performance.navigation.TYPE_BACK_FORWARD) { console.log('As per API lv1, this page is load from back/forward'); } else { console.log('This is normal page load'); @if (Session::has('sweet_alert.alert')) swal( {!! Session::get('sweet_alert.alert') !!} ); {{ Session::forget('sweet_alert.alert') }} // This will forget the alert data after displaying it :) @endif } } else { console.log("Unfortunately, your browser doesn't support this API"); } </script>
> 
> thank you man, your code works on me

if works on this package please share your code with me

**Comment by anggayudap** [comment]
> @anggayudap works for "uxweb/ sweet-alert" or this package ?
> if works on this package please share your code with me


on this package, I just put your js code at the bottom before the closing body tag.
please take a look this picture : [https://prnt.sc/U-89ljC_GRGL](url)

**Comment by karim20160300** [comment]
> > > this solution was working with me at "uxweb/ sweet-alert" i have tried to makes it fit with this package but i can not so i post it here maybe any one else can make it fit @realrashid
> > > <script> if (window.performance) { var navEntries = window.performance.getEntriesByType('navigation'); if (navEntries.length > 0 && navEntries[0].type === 'back_forward') { console.log('As per API lv2, this page is load from back/forward'); } else if (window.performance.navigation && window.performance.navigation.type == window.performance.navigation.TYPE_BACK_FORWARD) { console.log('As per API lv1, this page is load from back/forward'); } else { console.log('This is normal page load'); @if (Session::has('sweet_alert.alert')) swal( {!! Session::get('sweet_alert.alert') !!} ); {{ Session::forget('sweet_alert.alert') }} // This will forget the alert data after displaying it :) @endif } } else { console.log("Unfortunately, your browser doesn't support this API"); } </script>
> > 
> > 
> > thank you man, your code works on me
> 
> if works on this package please share your code with me
what is the package version which worked with this solution i am running v6 but the same solution not working for me

**Comment by anggayudap** [comment]
> > 
> > 
> > if works on this package please share your code with me
> > what is the package version which worked with this solution i am running v6 but the same solution not working for me

my project using laravel 8.xx and my composer.json dependency using `"realrashid/sweet-alert": "^5.1",`
here my full blade code:
https://pastebin.com/eW3C9BLB

**Comment by karim20160300** [comment]
> > > if works on this package please share your code with me
> > > what is the package version which worked with this solution i am running v6 but the same solution not working for me
> 
> my project using laravel 8.xx and my composer.json dependency using `"realrashid/sweet-alert": "^5.1",` here my full blade code: https://pastebin.com/eW3C9BLB

Thank you for sharing this with me, the issue has been resolved once downgrading from v6 to ^5.1 with the same solution

**Comment by theizekry** [comment]
try to use this middleware 

```
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Http\RedirectResponse;
use Illuminate\Http\Request;
use Illuminate\Http\Response;

class NoCacheHeaderMiddleware
{
    /**
     * Handle an incoming request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @param  \Closure(\Illuminate\Http\Request): (Response|RedirectResponse)  $next
     * @return Response|RedirectResponse
     */
    public function handle(Request $request, Closure $next)
    {
        $response = $next($request);

        $response->header('Cache-Control', 'no-cache, no-store, must-revalidate');
        $response->header('Pragma', 'no-cache');
        $response->header('Expires', '0');

        return $response;
    }
}

```

Please be noticed that This middleware is not inherently dangerous, but it does disable caching for all routes where it is applied. This can have performance implications, as disabling caching can lead to increased server load and slower response times. Additionally, it may not be appropriate to disable caching for all routes, depending on the specific needs of your application.

In general, it's a good practice to use caching in order to improve performance and reduce server load. However, there may be situations where you need to disable caching, such as when dealing with sensitive data that should not be cached by the browser.

If you decide to use this middleware, it's important to carefully consider the performance implications and to make sure that it is applied only where necessary. It's also a good idea to periodically review your caching strategy to make sure that it is still meeting the needs of your application.
